### ISSUE TITLE
EVE Online hangs

### ISSUE THREAD

**Issue by sdack** [doitsujin/dxvk#341]
The game EVE Online by CCP runs with DXVK, but after a couple of minutes hangs for no apparent reason. The rendered graphics look all right and the frame rate is also good.

GPU: Nvidia GTX960
Driver Nvidia 396.24
DXVK 0.42 (binary)
WINE 3.7 development (binary from WINEHQ)

The DXVK HUD is showing, but the *_d3d11.log is empty. The *_dxgi.log shows only one line:

`err:   DxvkInstance::getExtensions: Extension VK_KHR_surface not supported`

I'm not sure if I can run apitrace with it at the moment, because the game is started by a launcher application and seems to require some crypto keys. I'll look into it when I have some more time. I'm happy I got it running.

If anyone has got more information then please post them here. I'll also keep adding more once I have something.

Thank you.

**Comment by doitsujin** [comment]
Are there multiple executables for this game (e.g. a separate launcher)? If the D3D11 log is empty, the game doesn't actually *use* D3D11, but since you're getting the HUD that doesn't make sense either.

Same thing with the log message you're getting, it's basically telling you that Vulkan cannot be used, so if you are indeed getting the HUD, there must be some logs elsewhere.

**Comment by sdack** [comment]
Yes, there is a launcher application which starts the client application. The only log files I have found are:
```text
exefile_dxgi.log
exefile_d3d11.log
```
The client application is called `exefile.exe`, hence the file names. I assume the name is prepended by DXVK.

The HUD does not show for the launcher nor during the startup phase of the game. Only once the graphics render does the HUD show. The game also has a few visual difference based on whether it is using DX9 or DX11, and it is showing the DX11 graphics (i.e. brighter interiors lights and on the outisde one can see tiny little spaceships arriving at or leaving a station).

It is a 64-bit game and thus a mix of 32- and 64-bit libraries. WINE was installed with both i386 and amd64 packages, MESA drivers for Vulkan have both been installed for i386 and amd64, and also the Nvidia driver 396.24 installed the 32-bit compatibility libraries.

I've also installed the VulkanSDK 1.1.73 and ran the cube.exe demo, which brings up a rotating cube.

So I'm pretty confident that it is indeed working, only then it hangs. The hangs appear to occur faster without vsync. Using the new Vulkan compiler and turning it off both makes no difference.

**Comment by SveSop** [comment]
@sdack The launcher itself does not need to be a "graphical" application, thus no log is generated, and it will only happen once the game .exe is run. 

VulkanSDK should have both a cube.exe and a cube32.exe. Make sure both are working tho + ofc verify that your wine install has the correct d3d11.dll and dxgi.dll from DXVK in both system32 (64bit) and sysWOW64 (32bit).

Have not personally tried EVE, but where is the exefile.exe located? Same folder as the launcher? (Found some references to EVE/sharedcache/servername/bin... Afaik the .log files will be put in the same folder as the actual .exe binary)

**Comment by varris1** [comment]
Can you enable vsync or any other fps lock method and see if it still freezes? ESO and Warframe both have the same issue and they got fixed by enabling vsync in those games.

**Comment by sdack** [comment]
Sorry, was in a hurry and accidentally closed the issue.

The game is an MMO with free-to-play content. One can get it here: https://community.eveonline.com/support/download/ . It doesn't need to download the entire 8GB, but one can start playing after a few hundred MBs or so. Just in case anyone wants to give it a try.

@SveOps: I was only answering to doitsujin. Don't worry about it.

@varris1: I've alreadly mentioned vsync above. And yes, it does seem to have a bit of an extending effect on it, but it will still hang and I'm far from calling it a fix.

I'm currently setting up a clean environment where everything is compiled with the same compiler. Using a mix of binaries on a Debian testing distro with custom kernel and other new stuff isn't the savest way to approach this. So I'll be busy for a while before I make a better attempt. Also a new Nvidia driver is out and I'll throw it in for a test later.

**Comment by sdack** [comment]
I've now build WINE from scratch and have also tested Nvidia's driver 396.18.08. It's the exact same result as before with WINE from WineHQ and 396.24 and 396.18.05. Eventually the game hangs.

One thing I've noticed in addition to the hangs is the VSync control from within the game not working correctly. The game allows for the following settings:

* Interval Immediate - meaning VSync is off
* Interval One - 60fps
* Interval Two - 30fps
* Interval Three - 20fps
* Interval Four - 12.5fps

These settings work for DX9 and DX11 over OpenGL. With DXVK however do they all result in only a 60fps cap, meaning, "Interval Two" for example produces 60fps instead of the expected 30fps.

**Comment by sdack** [comment]
Update:

Nvidia 396.18.11 + WINE 3.8 + DXVK 0.51 still hangs.

**Comment by sdack** [comment]
I also now have log files with some content. Due to long repeated messages did I truncate duplicate lines and replaced them with "...". What stands out is:

```
err:   DxvkDevice: waitForIdle: Operation failed
err:   DxvkSubmissionQueue: Failed to sync fence: VK_ERROR_DEVICE_LOST
```

*_d3d11.log:
```text
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
err:   D3D11: Unsupported format: 0
err:   D3D11: Unsupported format: 66
err:   D3D11: Unsupported format: 89
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
warn:  D3D11Device::GetDeviceRemovedReason: Stub
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11Device: No matching border color found for (0.498039,0.498039,0,0)
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
...
... (repeats more than 30,000 times)
...
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
...
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
```
*_dxgi.log
```
info:  Enabled instance layers:
info:  Enabled instance extensions:
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
info:  GeForce GTX 960:
info:    Driver: 396.18.11
info:    Vulkan: 1.1.73
info:    Memory Heap[0]: 
info:      Size: 2048 MiB
info:      Flags: 0x1
info:      Memory Type[7]: Property Flags = 0x1
info:      Memory Type[8]: Property Flags = 0x1
info:    Memory Heap[1]: 
info:      Size: 11944 MiB
info:      Flags: 0x0
info:      Memory Type[0]: Property Flags = 0x0
info:      Memory Type[1]: Property Flags = 0x0
info:      Memory Type[2]: Property Flags = 0x0
info:      Memory Type[3]: Property Flags = 0x0
info:      Memory Type[4]: Property Flags = 0x0
info:      Memory Type[5]: Property Flags = 0x0
info:      Memory Type[6]: Property Flags = 0x0
info:      Memory Type[9]: Property Flags = 0x6
info:      Memory Type[10]: Property Flags = 0xe
warn:  Vulkan extension VK_EXT_vertex_attribute_divisor not supported
warn:  Vulkan extension VK_KHR_maintenance2 not supported
info:  Enabled device extensions:
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_maintenance1
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
info:  Enabled instance layers:
info:  Enabled instance extensions:
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
info:  GeForce GTX 960:
info:    Driver: 396.18.11
info:    Vulkan: 1.1.73
info:    Memory Heap[0]: 
info:      Size: 2048 MiB
info:      Flags: 0x1
info:      Memory Type[7]: Property Flags = 0x1
info:      Memory Type[8]: Property Flags = 0x1
info:    Memory Heap[1]: 
info:      Size: 11944 MiB
info:      Flags: 0x0
info:      Memory Type[0]: Property Flags = 0x0
info:      Memory Type[1]: Property Flags = 0x0
info:      Memory Type[2]: Property Flags = 0x0
info:      Memory Type[3]: Property Flags = 0x0
info:      Memory Type[4]: Property Flags = 0x0
info:      Memory Type[5]: Property Flags = 0x0
info:      Memory Type[6]: Property Flags = 0x0
info:      Memory Type[9]: Property Flags = 0x6
info:      Memory Type[10]: Property Flags = 0xe
warn:  Vulkan extension VK_EXT_vertex_attribute_divisor not supported
warn:  Vulkan extension VK_KHR_maintenance2 not supported
info:  Enabled device extensions:
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_maintenance1
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
warn:  Vulkan extension VK_EXT_vertex_attribute_divisor not supported
warn:  Vulkan extension VK_KHR_maintenance2 not supported
info:  Enabled device extensions:
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_maintenance1
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
warn:  DXGI: MakeWindowAssociation: Ignoring flags
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  1920x1080
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  32x32
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  1920x1080
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  32x32
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  1920x1080
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  32x32
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  1920x1080
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  32x32
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  1920x1080
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_IMMEDIATE_KHR
  Buffer size:  1920x1080
err:   DxvkDevice: waitForIdle: Operation failed
err:   DxvkSubmissionQueue: Failed to sync fence: VK_ERROR_DEVICE_LOST
err:   DxvkSubmissionQueue: Failed to sync fence: VK_ERROR_DEVICE_LOST
err:   DxvkSubmissionQueue: Failed to sync fence: VK_ERROR_DEVICE_LOST
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_IMMEDIATE_KHR
  Buffer size:  32x32
err:   DxvkDevice: waitForIdle: Operation failed
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
...
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_IMMEDIATE_KHR
  Buffer size:  1920x1080
err:   DxvkDevice: waitForIdle: Operation failed
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_IMMEDIATE_KHR
  Buffer size:  32x32
err:   DxvkDevice: waitForIdle: Operation failed
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_IMMEDIATE_KHR
  Buffer size:  1920x1080
err:   DxvkDevice: waitForIdle: Operation failed
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
...
err:   DxvkDevice: Command buffer submission failed: VK_ERROR_DEVICE_LOST
```

**Comment by doitsujin** [comment]
Can you run the game with `VK_INSTANCE_LAYERS=VK_LAYER_LUNARG_standard_validation` and capture the result? `VK_ERROR_DEVICE_LOST` usually indicates that something illegal is going on on the Vulkan side.

**Comment by sdack** [comment]
Here is what I get:

WINE itself with normal debug channels dumps out a large amount of the following messages:

```text
...
    Objects: 1
       [0] 0x19d20e, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d20e (Type = 23) | Descriptor set 0x19d20e bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #11 at global descriptor index 11 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d20f bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #5 at global descriptor index 5 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d20f, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d20f (Type = 23) | Descriptor set 0x19d20f bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #5 at global descriptor index 5 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d212 bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #7 at global descriptor index 7 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d212, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d212 (Type = 23) | Descriptor set 0x19d212 bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #7 at global descriptor index 7 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d21a bound as set #0 encountered the following validation error at vkCmdDraw() time: Descriptor in binding #9 at global descriptor index 9 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d21a, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d21a (Type = 23) | Descriptor set 0x19d21a bound as set #0 encountered the following validation error at vkCmdDraw() time: Descriptor in binding #9 at global descriptor index 9 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d26a bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #11 at global descriptor index 11 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d26a, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d26a (Type = 23) | Descriptor set 0x19d26a bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #11 at global descriptor index 11 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d26b bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #11 at global descriptor index 11 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d26b, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d26b (Type = 23) | Descriptor set 0x19d26b bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #11 at global descriptor index 11 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d26c bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #5 at global descriptor index 5 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d26c, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d26c (Type = 23) | Descriptor set 0x19d26c bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #5 at global descriptor index 5 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d26f bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #7 at global descriptor index 7 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d26f, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d26f (Type = 23) | Descriptor set 0x19d26f bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #7 at global descriptor index 7 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d277 bound as set #0 encountered the following validation error at vkCmdDraw() time: Descriptor in binding #9 at global descriptor index 9 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d277, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d277 (Type = 23) | Descriptor set 0x19d277 bound as set #0 encountered the following validation error at vkCmdDraw() time: Descriptor in binding #9 at global descriptor index 9 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d2c8 bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #11 at global descriptor index 11 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d2c8, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d2c8 (Type = 23) | Descriptor set 0x19d2c8 bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #11 at global descriptor index 11 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d2c9 bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #11 at global descriptor index 11 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d2c9, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d2c9 (Type = 23) | Descriptor set 0x19d2c9 bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #11 at global descriptor index 11 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d2ca bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #5 at global descriptor index 5 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d2ca, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d2ca (Type = 23) | Descriptor set 0x19d2ca bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #5 at global descriptor index 5 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d2cd bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #7 at global descriptor index 7 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d2cd, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d2cd (Type = 23) | Descriptor set 0x19d2cd bound as set #0 encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #7 at global descriptor index 7 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
(null)(ERROR / SPEC): msgNum: 61 - Descriptor set 0x19d2d5 bound as set #0 encountered the following validation error at vkCmdDraw() time: Descriptor in binding #9 at global descriptor index 9 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
    Objects: 1
       [0] 0x19d2d5, type: 23, name: (null)
Validation(ERROR): msg_code: 61: Object: 0x19d2d5 (Type = 23) | Descriptor set 0x19d2d5 bound as set #0 encountered the following validation error at vkCmdDraw() time: Descriptor in binding #9 at global descriptor index 9 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
```
The *_d3d11.log file looks pretty much like the one before:
```text
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
err:   D3D11: Unsupported format: 0
err:   D3D11: Unsupported format: 66
err:   D3D11: Unsupported format: 89
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
warn:  D3D11Device::GetDeviceRemovedReason: Stub
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11Device: No matching border color found for (0.498039,0.498039,0,0)
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
warn:  D3D11DeviceContainer::QueryInterface: Unknown interface query
warn:  6543dbb6-1b48-42f5-ab82-e97ec74326f6
... (repeats about 13,000 times)
```

*_dxgi.log:
```text
info:  Enabled instance layers:
info:    VK_LAYER_LUNARG_standard_validation
info:  Enabled instance extensions:
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
info:  GeForce GTX 960:
info:    Driver: 396.18.11
info:    Vulkan: 1.1.73
info:    Memory Heap[0]: 
info:      Size: 2048 MiB
info:      Flags: 0x1
info:      Memory Type[7]: Property Flags = 0x1
info:      Memory Type[8]: Property Flags = 0x1
info:    Memory Heap[1]: 
info:      Size: 11944 MiB
info:      Flags: 0x0
info:      Memory Type[0]: Property Flags = 0x0
info:      Memory Type[1]: Property Flags = 0x0
info:      Memory Type[2]: Property Flags = 0x0
info:      Memory Type[3]: Property Flags = 0x0
info:      Memory Type[4]: Property Flags = 0x0
info:      Memory Type[5]: Property Flags = 0x0
info:      Memory Type[6]: Property Flags = 0x0
info:      Memory Type[9]: Property Flags = 0x6
info:      Memory Type[10]: Property Flags = 0xe
warn:  Vulkan extension VK_EXT_vertex_attribute_divisor not supported
warn:  Vulkan extension VK_KHR_maintenance2 not supported
info:  Enabled device extensions:
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_maintenance1
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
info:  Enabled instance layers:
info:    VK_LAYER_LUNARG_standard_validation
info:  Enabled instance extensions:
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
info:  GeForce GTX 960:
info:    Driver: 396.18.11
info:    Vulkan: 1.1.73
info:    Memory Heap[0]: 
info:      Size: 2048 MiB
info:      Flags: 0x1
info:      Memory Type[7]: Property Flags = 0x1
info:      Memory Type[8]: Property Flags = 0x1
info:    Memory Heap[1]: 
info:      Size: 11944 MiB
info:      Flags: 0x0
info:      Memory Type[0]: Property Flags = 0x0
info:      Memory Type[1]: Property Flags = 0x0
info:      Memory Type[2]: Property Flags = 0x0
info:      Memory Type[3]: Property Flags = 0x0
info:      Memory Type[4]: Property Flags = 0x0
info:      Memory Type[5]: Property Flags = 0x0
info:      Memory Type[6]: Property Flags = 0x0
info:      Memory Type[9]: Property Flags = 0x6
info:      Memory Type[10]: Property Flags = 0xe
warn:  Vulkan extension VK_EXT_vertex_attribute_divisor not supported
warn:  Vulkan extension VK_KHR_maintenance2 not supported
info:  Enabled device extensions:
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_maintenance1
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
warn:  Vulkan extension VK_EXT_vertex_attribute_divisor not supported
warn:  Vulkan extension VK_KHR_maintenance2 not supported
info:  Enabled device extensions:
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_maintenance1
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
warn:  DXGI: MakeWindowAssociation: Ignoring flags
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  1920x1080
info:  DxgiVkPresenter: Recreating swap chain: 
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_IMMEDIATE_KHR
  Buffer size:  1920x1080
```

**Comment by sdack** [comment]
Seeing how this are quite a lot of message did I capture it into a file.

[log.gz](https://github.com/doitsujin/dxvk/files/2000786/log.gz)

**Comment by doitsujin** [comment]
The messages about certain extensions or features not being enabled are complete nonsense (DXVK even logs the extension in question as being enabled!) and I'm really not sure why the validation layers are complaining about that, and most of the other messages are caused by game bugs that don't really matter in practice.

But this one is nasty:
```
Validation(ERROR): msg_code: 61: Object: 0x126f7 (Type = 23) | Descriptor set 0x126f7 bound as set #0 encountered the following validation error at vkCmdDraw() time: Descriptor in binding #9 at global descriptor index 9 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
```
This might be the cause of this issue, but it's going to be extremely hard to work around this. Unless I run into other games doing the same thing I doubt I'll be fixing this any time soon.

The problem here is that the game binds a non-multisampled texture to a resource slot that expects a multisampled texture. Not sure if this is even allowed in Dx11 (might be *another* game bug), but this game expects it to work.

**Comment by doitsujin** [comment]
> Do you think you could do something like this?

Yes, but that's probably incorrect. I'll have to run some tests on Windows for that though.

**Comment by rafalcieslak** [comment]
If you're looking for more ways to trigger this error, then it may be useful to know that Witcher 3 running with `VK_INSTANCE_LAYERS=VK_LAYER_LUNARG_standard_validation` on `dxvk 0.51` triggers **a lot** of:
```
DS(ERROR): object: 0x420587f type: 23 location: 1123 msgCode: 61: Descriptor set 0x420587f encountered the following validation error at vkCmdDraw() time: Descriptor in binding #1 at global descriptor index 1 requires bound image to have multiple samples, but got VK_SAMPLE_COUNT_1_BIT.
```

**Comment by sdack** [comment]
Update: DXVK 0.52 + Nvidia 396.18.11 still produces hangs and the game's VSync settings behave wrong as before.

**Comment by DestroyFX** [comment]
I just decided to test DXVK again (been a while)

AMD with RADV.
* Have only Interval immediate and Interval 1 (use to have Interval 2, 3... but it's gone)
* Vsync work fine. 100fps in station, 130 fps outside station in 3840x2160 with everything to high except post processing is OFF (use to leave that OFF for d3d9 galium nine graphics bug)
* Still did not encounter any crash. Could it be that post processing? or Nvidia related?

**Comment by sdack** [comment]
Update: WINE 3.9 + Nvidia 396.24.02 + DXVK 0.52 (commit: b78130defd4b3b19543ed4ce1244e607131814dc) leads to hangs and also same VSync issue as before.

**Comment by varris1** [comment]
Did you try to disable "Allow Flipping" in nvidia-settings? It seems to have fixed my freezing issues in TESO. Maybe I'll help you as well.

**Comment by sdack** [comment]
@varris1 I've tried it and it has indeed stopped the hangs. Much appreciated. I hadn't expected an OpenGL setting having an effect on Vulkan tbh.

So this fixes the hangs and I've ran some tests, including an auto-pilot run to Jita and orbiting outside of Jita station. Still no hangs.

What remains are the VSync issues, which are still there and now that I could test some more have I seen some visual artifacts, too. One being flicking on the in-game HUD (not the HUD of DXVK) where the speed meter and the outer circle shows flickering. Also the HUD of the first-person camera shows fractured graphics.

Still, this is all in all very playable. I'll see if I can get test some more and also make some screenshots later.

This was done with: WINE 3.9 + Nvidia 296.24.02 + DXVK 0.53

**Comment by doitsujin** [comment]
I'm aware that Vsync with intervals other than one frame doesn't work, it was implemented in DXVK at some point but was removed due to general issues with the presentation code. I can add it back though.

As for the visual issues, I need an apitrace to debug those. Feel free to open a new issue about that
